### DISCUSSION TITLE
One url per locale? Multi-domain deployment.

### DISCUSSION THREAD

**Discussion by ahtremblay** [facebook/docusaurus#7854] [General]
Hello, it does not seem possible to set one url for each locale.

For instance, I may wish to deploy the `fr` local to `https://www.bonjour.com` and the `en` locale to `https://www.hello.com`, but `docusaurus.config.js` only allows to specify a single `url` for the whole project. Is there a workaround?

The relevant parts of my config file are:

```
const config = {
  url: 'https://www.hello.com',
  i18n: {
    defaultLocale: 'en',
    locales: ['en', 'fr']
  },
```

I compile using a build.sh script as follows:

```
#!/bin/bash
rm -rf build
npm run build -- --locale fr --out-dir build/fr
npm run build -- --locale en --out-dir build/en
```

However, the links in the generated files all point to `https://www.hello.com` regardless of the locale. How can I make the url for the `fr` locale point to `https://www.bonjour.com`? The sitemap.xml also contains `https://www.hello.com` for the `fr` locale.

**Comment by slorber** [comment]
@ahtremblay we should be able to customize the config according to the current locale, but don't have a good API for that ATM, and the original design of the config does not really permit to add it easily without a breaking change or a weird API design.

The only workaround I can suggest is to run multiple builds using different environment variables.

```
#!/bin/bash
rm -rf build
CURRENT_DOCUSAURUS_LOCALE=fr npm run build -- --locale fr --out-dir build/fr
CURRENT_DOCUSAURUS_LOCALE=en npm run build -- --locale en --out-dir build/en
```

And then use `process.env.CURRENT_DOCUSAURUS_LOCALE` in the config file to decide on different configs.

```js
const config = {
  url: process.env.CURRENT_DOCUSAURUS_LOCALE === "fr" ? 'https://www.bonjour.com' : 'https://www.hello.com',
  i18n: {
    defaultLocale: 'en',
    locales: ['en', 'fr']
  },
}
```

At best we'll provide this env variable automatically for you, or find a way to make the current locale available in the config file. The weird issue is that we have to load the config file once first without any "current locale", just because we actually need to read the i18n config  and default locale 😅 )

Please let me know if this works for you.

Note I'd like to move the Docusaurus i18n sites to subdomains later, just for the sake of demonstrating this setup is technically possible.
